video/out/vo_dmabuf_wayland.c

```diff
@@ -73,19 +73,25 @@ static bool vaapi_dmabuf_importer(struct mp_image *src, struct wlbuf_pool_entry*
                                   struct zwp_linux_buffer_params_v1 *params)
 {
     struct priv *p = entry->vo->priv;
-    VADRMPRIMESurfaceDescriptor desc;
+    VADRMPRIMESurfaceDescriptor desc = { 0 };
     bool dmabuf_imported = false;
     /* composed has single layer */
     int layer_no = 0;
+    uint32_t drm_format;
     VAStatus status = vaExportSurfaceHandle(p->display, entry->key, VA_SURFACE_ATTRIB_MEM_TYPE_DRM_PRIME_2,
                                             VA_EXPORT_SURFACE_COMPOSED_LAYERS | VA_EXPORT_SURFACE_READ_ONLY, &desc);
-    if (status == VA_STATUS_ERROR_INVALID_SURFACE) {
-        MP_VERBOSE(entry->vo, "VA export to composed layers not supported.\n");
-    } else if (!vo_wayland_supported_format(entry->vo, desc.layers[0].drm_format, desc.objects[0].drm_format_modifier)) {
+
+    if (status == VA_STATUS_ERROR_INVALID_SURFACE)
+        MP_VERBOSE(entry->vo, "vaExportSurfaceHandle: composed layers not supported.\n");
+    if (!CHECK_VA_STATUS(entry->vo, "vaExportSurfaceHandle()"))
+        return false;
```

Comment:
I think this is still a bit weird for me as it is checking for an error state just before generic error checking. While it would reorder the logging, moving the logging within the if would more clearly show the logical block into which both of these go into ("if the export did not succeed").

So with this state it would be (based on [this](https://github.com/intel/libva/blob/2579eb0f77897dc01a02c1e43defc63c40fd2988/va/va.c#L634-L635))
```
[v] vaExportSurfaceHandle: composed layers not supported.
[e] vaExportSurfaceHandle() failed ("invalid VASurfaceID")
```
and with it being within the if it would just get reordered (which doesn't seem to be too bad for me).